tests/mmm/test_utils.py

```diff
@@ -344,3 +346,33 @@ def test_create_new_spend_data(
         new_spend_data,
         np.array(expected_result),
     )
+
+
+def test_create_new_spend_data_value_errors() -> None:
+    with pytest.raises(
+        ValueError, match="spend_leading_up must be the same length as the spend"
+    ):
+        create_new_spend_data(
+            spend=np.array([1, 2]),
+            adstock_max_lag=2,
+            one_time=True,
+            spend_leading_up=np.array([3, 4, 5]),
+        )
+
+
+@pytest.mark.parametrize(
+    argnames="distribution_dict, expected",
+    argvalues=[
+        ({"dist": "Normal"}, pm.Normal),
+        ({"dist": "Gamma"}, pm.Gamma),
+        ({"dist": "StudentT"}, pm.StudentT),
+    ],
+    ids=["Normal", "Gamma", "StudentT"],
+)
+def test_get_distribution_from_dict(distribution_dict, expected):
+    assert _get_distribution_from_dict(distribution_dict) == expected
+
+
+def test_get_distribution_from_dict_value_error():
+    with pytest.raises(ValueError):
+        _get_distribution_from_dict({"dist": "InvalidDistribution"})
```

Comment:
There are some tests in tests/mmm/test_delayed_saturation_mmm.py for this 
Let's delete those and go with theses

Reply:
looking into it, it seems we have duplicated code anyway https://github.com/pymc-labs/pymc-marketing/pull/738/commits/e6067e04a6a68052a96c28cc6cabbc3de8fc088d